pallets/sp-mvm/src/lib.rs

```diff
@@ -220,17 +224,37 @@ pub mod pallet {
             module_bc: Vec<u8>,
             gas_limit: u64,
         ) -> DispatchResultWithPostInfo {
-            let who = ensure_signed(origin)?;
-            debug!("executing `publish` with signed {:?}", who);
+            // Allows to update Standard Library if root.
+            let (sender, signer) = match T::UpdaterOrigin::ensure_origin(origin.clone()) {
+                Ok(_) => {
+                    debug!("executing `publish module` with root");
+                    (CORE_CODE_ADDRESS, sudo::Pallet::<T>::key())
+                }
+                Err(_) => {
+                    let signer = ensure_signed(origin)?;
+                    debug!("executing `publish module` with signed {:?}", signer);
+                    (addr::account_to_account_address(&signer), signer)
+                }
+            };
+            debug!("executing `publish` with signed {:?}", sender);
 
             // Publish module.
-            let vm_result = Self::raw_publish_module(&who, module_bc, gas_limit, false)?;
+            let vm = Self::get_vm()?;
+            let gas = Self::get_move_gas_limit(gas_limit)?;
+
+            let tx = {
+                let sender = addr::account_to_bytes(&sender);
+                debug!("converted sender: {:?}", sender);
+                ModuleTx::new(module_bc, AccountAddress::new(sender))
+            };
+
+            let vm_result = vm.publish_module(gas, tx, false);
```

Comment:
Would be great use `raw_publish_module` here and so move this root-determining and address conversion into the `raw_publish_module`.
Also you can make `raw_publish_module` accept `origin: OriginFor<T>` if needed.

Reply:
Or just define special method for it! And name it `ensure_and_convert`.
_This is the same algorithm as in [`publish_package` method](https://github.com/pontem-network/pontem/pull/170/files#diff-4e5930d2e2ba9c4d86fb707cc5d1eb7aef9839019ea8e6f3125310aa710557cbL255-R287).